### PR TITLE
Add start price to bid_label for future auctions

### LINKED ISSUE #1292: Admin and Vendor messaging - Auction show page header - Starting price for auctions that are coming soon
As a vendor or a visitor, I want to see the starting price of auctions that are coming soon so I can decide whether I want to bid on that auction.

- [x] **Vendor/visitor sees starting price in header of show page for auctions that are coming soon**
- Given I am a visitor to the micro-purchase site
- And there is a reverse auction
- And that auction is published
- And that auction has not yet started
- And the auction has no bids
- When I go the auction show page for that auction (micropurchase.18f.gov/auctions/[id]) or (micropurchase.18f.gov/admin/auctions/[id])
- Then I should see a section in the header labeled "Starting price:" 
- And in that section I should see the starting price of that auction

## Notes
Use the styles referenced below (or as close as you can get)

screen shot 2016-09-20 at 4 45 37 pm

### LINKED ISSUE #1293: Admin and vendor messaging - Auction show page - Winning bid for closed auctions
As an admin or a vendor, I want to quickly see the winning bid for any auction, so if I am viewing the show page of an auction, the winning bid should be shown to me.

- [x] **Admin and Vendor/visitor sees winning bid and winning vendor name in header of show page for closed auctions**
- Given I am a visitor to the micro-purchase site
- And there is an auction (reverse or sealed-bid)
- And that auction has ended
- And the auction has a winning bid
- When I go the auction show page for that auction (micropurchase.18f.gov/auctions/[id]) or (micropurchase.18f.gov/admin/auctions/[id])
- Then I should see a section in the header labeled "Winning bid:" 
- And in that section I should see the winning bid for that auction
- And I should see the winning vendor name (or GitHub name) in parentheses after the bid

## Notes
Use the styles referenced below (or as close as you can get)

screen shot 2016-09-20 at 4 45 37 pm

### PR DESCRIPTION
**Pull request by harrisj**
Fixes #1292 
Fixes #1293 

In addition, I decided to add the bid_label to admin/auctions/:id show since it's required for this pull request, but I also supported some other status messages.

### CODE REVIEW

**Comment by harrisj** [comment]
I'd like to let the CodeClimate thing stand for now

File: app/view_models/admin/auction_show_view_model.rb [hunk 1]
```diff
@@ -77,8 +77,48 @@ def html_summary
...
   end
 
+  def bid_label
+    if over? && auction.bids.any?
+      "Winning bid (#{lowest_bidder_name}): #{highlighted_bid_amount_as_currency}"
+    elsif auction.bids.any?
+      "Current bid: #{highlighted_bid_amount_as_currency}"
+    elsif future?
+      "Starting price: #{Currency.new(auction.start_price)}"
+    else
```
Comment (jessieay) [inline app/view_models/admin/auction_show_view_model.rb]: maybe add a comment here to explain when we'd get here?

File: app/view_models/admin/auction_show_view_model.rb [hunk 2]
```diff
@@ -77,8 +77,48 @@ def html_summary
     MarkdownRender.new(auction.summary).to_s
   end
 
+  def bid_label
```
Comment (jessieay) [inline app/view_models/admin/auction_show_view_model.rb]: test for this?

**Comment by harrisj** [comment]
Okay, I added some tests. Thank you.

File: spec/view_models/auction_show_view_model_spec.rb [hunk 3]
```diff
@@ -24,4 +24,51 @@
...
+      user = create(:user)
+
+      view_model = AuctionShowViewModel.new(auction: auction, current_user: user)
+
+      expect(view_model.bid_label).to eq("Winning bid (#{name}): #{Currency.new(bid.amount)}")
+    end
+
+    it "should show the user's bid when the user has bid" do
+      auction = create(:auction, :with_bids)
+      bid = auction.bids[-1]
```
Comment (jessieay) [inline spec/view_models/auction_show_view_model_spec.rb]: why this bid? using `-1` index feels a bit mysterious...

File: app/view_models/admin/auction_show_view_model.rb (hunk 1 above)
Comment (jessieay) [inline app/view_models/admin/auction_show_view_model.rb]: thoughts?

**Comment by jessieay** [comment]
left two small final comments! looking good

File: spec/view_models/auction_show_view_model_spec.rb (hunk 3 above)
Comment (harrisj) [inline spec/view_models/auction_show_view_model_spec.rb]: Oh I just wanted to not use the last bid, but all I had

File: app/view_models/admin/auction_show_view_model.rb [hunk 4]
```diff
@@ -77,8 +77,48 @@ def html_summary
...
+
+  def highlighted_bid_amount_as_currency
+    Currency.new(rules.highlighted_bid(current_user).amount).to_s
+  end
+
+  def over?
+    bidding_status.over?
+  end
+
+  def available?
```
Comment (harrisj) [inline app/view_models/admin/auction_show_view_model.rb]: No we aren't. You're right!

**Pull request merged**